boards/shields/x_nucleo_iks02a1/x_nucleo_iks02a1_shub.overlay

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2020 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&arduino_i2c {
+
+	iis2dlpc@19 {
+		compatible = "st,iis2dlpc";
+		reg = <0x19>;
+		drdy-gpios =  <&arduino_header 4 GPIO_ACTIVE_HIGH>; /* A4 - INT2 */
+		label = "IIS2DLPC";
+	};
+
+	ism330dhcx@6b {
+		compatible = "st,ism330dhcx";
+		reg = <0x6b>;
+		drdy-gpios =  <&arduino_header 11 GPIO_ACTIVE_HIGH>; /* D5 - INT2 */
+		label = "ISM330DHCX";
+	};
+};
```

Comment:
I know we've split overlay files this way before.
But is that really required? these are the same nodes, at least from device tree point of view.
The only difference is how they'll be configured at the Kconfig level. So, a unique overlay file should be fine IMHO (together with _mic)

Reply:
ok, I think it can be done.
I'll take a look at it right now.

Reply:
OOops, just a couple of doubts that I found immediately:

We have 3 different configurations:
  - standard (Mode 1)
  - sensorhub (Mode 2)
  -  microphone

1. At least the standard and sensorhub solution should have two different dts. This because the IIS2MDC is visible only in standard mode. In sensorhub mode is infact attached to ISM330HCX sensorhub, so it should not be declared in dts. I guess that using a single overlay file would lead to confusion.
2. If I put the i2s part in standard overlay (for example) and use it for microphone sample, I need to be add arduino_i2s for othe board as well, like for example nucleo_f401re, otherwise the sensor standard mode can be tried on f411re only. What do you think?

Reply:
OK, I solved #2 adding arduino_i2s to f401re as well. Other boards may be extended in the future.

For #1 I propose to use two separate overlay files (I removed the mic one). Another possibility is to use ifdefs in the overlay. What do you think?

Reply:
> At least the standard and sensorhub solution should have two different dts. This because the IIS2MDC is visible only in standard mode. In sensorhub mode is infact attached to ISM330HCX sensorhub, so it should not be declared in dts. I guess that using a single overlay file would lead to confusion.

I'm not quite in line here. dts is supposed to describe available hardware and this is the same shield you are using in both cases, there is no reason to use 2 different hardware descriptions.
To my understanding hub is a software configuration of the available hardware, this should be done in Kconfig and btw this is already what you are doing with `CONFIG_LSM6DSO_SENSORHUB`.
Are you facing issues when using same overlay in both cases ?


> Another possibility is to use ifdefs in the overlay.

We can't use #ifdef in overlays as dts is processed before Kconfig

Reply:
> > At least the standard and sensorhub solution should have two different dts. This because the IIS2MDC is visible only in standard mode. In sensorhub mode is infact attached to ISM330HCX sensorhub, so it should not be declared in dts. I guess that using a single overlay file would lead to confusion.
> 
> I'm not quite in line here. dts is supposed to describe available hardware and this is the same shield you are using in both cases, there is no reason to use 2 different hardware descriptions.

But the shield can be configured using JP7. There are two modes: Mode1 (standard) and Mode2 (sensorhub). Once that you move that switch in one of the two positions the hardware is different. The connections are different and IIS2MDC is no longer accessible from MCU, but only from ISM330DHCX. So, two different hardwares requires two different DTS.

> To my understanding hub is a software configuration of the available hardware, this should be done in Kconfig and btw this is already what you are doing with `CONFIG_LSM6DSO_SENSORHUB`.

No, no. It is really a different h/w configuration. IIS2MDC is no longer connected to MCU I2C.

> Are you facing issues when using same overlay in both cases ?
> 
> > Another possibility is to use ifdefs in the overlay.
> 
> We can't use #ifdef in overlays as dts is processed before Kconfig
ok. So I don't see other options but having two dts (because we have two different h/w on the same shield)

Reply:
Just one more info here. IIS2MDC sensor is on I2C1 bus, while the other two are on I2C1, where the mcu connects to. 
Using JP7 you may connect I2C2 to I2C1 (Mode 1) so that also IIS2MDC is visible to mcu, or you may connect it to SCx/SDx of ISM330DHCX, so that is ONLY that sensor that can access it.
I find that using the same dts may result in a confusion.

Reply:
ok, in that case, that's indeed 2 different hardware combinations and 2 overlays are needed. Thanks for the clarification.
Maybe this is worth a comment directly in the overlay files though.

Reply:
Yes, sure, I'll add it. I will also review the board/shield/iks02a1 index.rst documentation in order to understand if the info are enough.

Reply:
Shield doc is clear on that point, but it wouldn't hurt adding a reminder in these overlay files

Reply:
done